### DISCUSSION TITLE
Help to run the Patrol examples

### DISCUSSION THREAD

**Discussion by cristianorsilva** [leancodepl/patrol#1039] [Q&A] [answered]
Hi!

I have tried to use Patrol in my projects, then I followed the instructions here (https://patrol.leancode.co/native/overview). However in all of them I got the error below:

**'package:flutter/src/foundation/binding.dart': Failed assertion: line 149 pos 12: '_debugInitializedType == null': is not true.**

I suspected that I was missing something in my projects, then I decided to run the patrol examples (https://github.com/leancodepl/patrol/tree/master/packages/patrol/example) but to my surprise, I got the same error: 

**Failed to load "/Users/cristiano/Downloads/patrol/patrol-master/packages/patrol/example/integration_test/open_app_test.dart": 'package:flutter/src/foundation/binding.dart': Failed assertion: line 149 pos 12: '_debugInitializedType == null': is not true.**

Am I missing some config in my setup? Any ideas about what is happening? 

Thanks in advance!

Below some information about my setup:

**flutter doctor -v:**

[✓] Flutter (Channel stable, 3.7.6, on macOS 13.2.1 22D68 darwin-arm64, locale en-BR)
    • Flutter version 3.7.6 on channel stable at /Users/cristiano/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 12cb4eb7a0 (3 days ago), 2023-03-01 10:29:26 -0800
    • Engine revision ada363ee93
    • Dart version 2.19.3
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/cristiano/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • ANDROID_HOME = /Users/cristiano/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.2.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] Connected device (3 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554 • android-arm64  • Android 13 (API 33) (emulator)
    • macOS (desktop)             • macos         • darwin-arm64   • macOS 13.2.1 22D68 darwin-arm64
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 110.0.5481.177

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

**patrol --version:**

patrol_cli v1.0.5

**Below the complete error when I tried to run the example 'open_app_test.dart':**

Testing started at 16:17 ...
Running Gradle task 'assembleDebug'...                             12.4s
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app-debug.apk...        1,181ms

dart:core                                                                                                        _AssertionError._throwNew
package:flutter/src/foundation/binding.dart 149:12                                                               new BindingBase
package:flutter_test/src/binding.dart                                                                            new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding
package:flutter_test/src/binding.dart                                                                            new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding
package:flutter_test/src/binding.dart                                                                            new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding
package:flutter_test/src/binding.dart                                                                            new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding&SemanticsBinding
package:flutter_test/src/binding.dart                                                                            new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding&SemanticsBinding&RendererBinding
package:flutter_test/src/binding.dart                                                                            new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding&SemanticsBinding&RendererBinding&PaintingBinding
package:flutter_test/src/binding.dart                                                                            new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding&SemanticsBinding&RendererBinding&PaintingBinding&WidgetsBinding
package:flutter_test/src/binding.dart                                                                            new _TestWidgetsFlutterBinding&BindingBase&SchedulerBinding&ServicesBinding&GestureBinding&SemanticsBinding&RendererBinding&PaintingBinding&WidgetsBinding&TestDefaultBinaryMessengerBinding
package:flutter_test/src/binding.dart                                                                            new TestWidgetsFlutterBinding
package:flutter_test/src/binding.dart                                                                            new LiveTestWidgetsFlutterBinding
package:integration_test/integration_test.dart                                                                   new IntegrationTestWidgetsFlutterBinding
package:patrol/src/binding.dart                                                                                  new PatrolBinding
package:patrol/src/binding.dart 84:7                                                                             new PatrolBinding.ensureInitialized
package:patrol/src/common.dart 52:39                                                                             patrolTest
Users/cristiano/Downloads/patrol/patrol-master/packages/patrol/example/integration_test/common.dart 22:3         patrol
Users/cristiano/Downloads/patrol/patrol-master/packages/patrol/example/integration_test/open_app_test.dart 15:3  main

Failed to load "/Users/cristiano/Downloads/patrol/patrol-master/packages/patrol/example/integration_test/open_app_test.dart": 'package:flutter/src/foundation/binding.dart': Failed assertion: line 149 pos 12: '_debugInitializedType == null': is not true.

**Comment by bartekpacia** [comment, marked as answer]
Hi! What command are you executing to run the tests?

Make sure to use [patrol test](https://patrol.leancode.co/cli-commands/test).

**Reply by cristianorsilva** [reply]
Hi! You got it!!! My mistake was trying to run only by the left green icon on the test, but when I ran the command 'patrol test --target integration_test/android_app_test.dart' it worked perfectly :)

Patrol examples working here. Now I will fix my ones, got some errors but I think I can handle :)

Thanks!!!

**Reply by bartekpacia** [reply]
Happy to help :) In case you encounter more problems, don't hesitate to create issues (or if you have general questions, create a discussion).
